zerver/lib/users.py

```diff
@@ -546,10 +548,13 @@ def get_raw_user_data(
         profiles_by_user_id = get_custom_profile_field_values(custom_profile_field_values)
 
     result = {}
+    client_gravatar_initial_value = client_gravatar
     for row in user_dicts:
+        client_gravatar = client_gravatar_initial_value
         if profiles_by_user_id is not None:
             custom_profile_field_data = profiles_by_user_id.get(row["id"], {})
-
+        if row["email_address_visibility"] != UserProfile.EMAIL_ADDRESS_VISIBILITY_EVERYONE:
+            client_gravatar = False
```

Comment:
Mutation of inputs like this is messy.  A cleaner way to do this loop would be to just pass to `format_user_row` the value:

client_gravatar_for_user=client_gravatar and row["email_address_visibility"] == UserProfile.EMAIL_ADDRESS_VISIBILITY_EVERYONE